### ISSUE TITLE
[GoldSource/CS][Feature] Kill & flash assist 

### ISSUE THREAD

**Issue by ArmynC** [ValveSoftware/halflife#2229]
There should be a feature like te CS:GO's Kill Assist (cvar set rewards) and also a flash assist (teammate kill an enemy flashed by you) with a little flash icon.

![Example](https://user-images.githubusercontent.com/16375271/56518503-53d48c80-6548-11e9-97a0-cbac215282b7.png)

[Code example](https://dev-cs.ru/resources/189/)

**Comment by djdallmann** [comment]
Would love this, not to fully derail but also having molotov/incendiary as a toggle, and perhaps better smokes.

**Comment by PM32** [comment]
I think it's better to focus on engine/gamedll improvement instead of requesting features, IMO.

**Comment by ArmynC** [comment]
> I think it's better to focus on engine/gamedll improvement instead of requesting features, IMO.

We just do the "tickets" and Valve reps. will choose according to their priorities :)

**Comment by simosalis** [comment]
You guys (ArmynC, perforatorRU, Splatt581 and a few others), should really stop spamming bug fixes & features request to valve, also knowing those are features that can easily be fixed via plugins, or already did with another CS binary.
Valve maybe performed some updates recently (as one major against security issues), mainly because this was important, and then, such problems were known from years and they didn't hung up to do it.

All of you probably got some hope due to the fact they worked again on updates, but you shouldn't expect much more, they did not make a lot of efforts to satisfy people requests from years, a ton of demands were just ignored and will continue to be.

It will be nice to have a correct official (from valve) development of those games, plus gold source engine itself, but some alternative binaries (regamedll) exist for this, where those changes will be added before valve will do or never will.

So you should only post most important issues (just my opinion, acting as an advice), or maybe, discuss together at making a global post for those bug fixes, properly structured.

**Comment by ArmynC** [comment]
As I said, nothing bad can happen if we post these issue tickets. Until now more issues were answered and implemented. In this manner even the most dangerous bugs can be solved via plugins, but not all of them (Admins) know about adding these patches. With these bigger or little bug-fixes & features (that can be easily part of the vanilla core), I wanted achieve the same effect. Rather than having nothing, is better to have more things from where the employees can choose what to do (or not). I didn't had a great hope but a community duty to show more things that Valve may be not aware of...

About global post I think is a good ideea but in GitHub environment I don't know if it is accepted :confused:

**Comment by GiovaniFerraroTrivelli** [comment]
I think if you want those new features, you should play CS:GO (or CS:S at least). The priorities are bugfixes and improvements (like hitbox), or simple additions that doesn't break gameplay or need full testing (burst indicator).

Keep in mind that there are so many custom mods on CS 1.6, and there is the risk that adding new features can break them (I know this is server owners problem, but like you said "_but not all of them (Admins) know about adding these patches_", fixing mods is worst for them).

**Comment by ArmynC** [comment]
Yeah, I reporter mostly bugs alongside 2-3 features (which mostly aren't <in this case anymore> in CS:GO)..

**Comment by simosalis** [comment]
Yes, ArmynC, there is basically almost nothing wrong at posting requests for bug fixes and new features, but this "spam" the issues for something that will not be considered, or something not important/trivial, which could be easily fixed via plugins (and already available, since those plugins exists, Numb author did a lot regarding those problems).
From that I know, there is no official "cs 1.7" in development, and even if we saw a valve's employee (mikela) registers over here, that doesn't mean we can expect more, however, it will be good for sure (if valve has decided to take care more of those old good games, then this guy more active on half-life 1 games).

About a global issues topic, separated in two categories (one for bug fixes, one for new features), a board like (https://github.com/s1lentq/ReGameDLL_CS/issues/185), could be made to handle all of that, and new requests could be posted at the suite (via simple post), then, due to their importance level, added to the board or not.

By the way, I've heard about a beta client version, but I'm unable to find a changelog of this, or I've missed it.
I don't use it, but I've seen somewhere else there is a modification in the scoreboard, where the money is now displayed. I personnaly think it would have been better to have the healh (even if already in % when aiming someone depending on mp_playerid value). Because except with a trading plugin, or, to help a friend by buying weapon for him, I think the health is more useful, in order to know who is the friend who needs assistance because he is close to die ("Need backup!" request from his radio).

Does someone have a changelog link (beta version only, not the standard one, which can be found in the HUB)?

**Comment by eat1k** [comment]
> Does someone have a changelog link (beta version only, not the standard one, which can be found in the HUB)?

There is no changelog. You can find all the fixes which have been done in Issues (where mikela-valve is assigned).

**Comment by voed** [comment]
@mikela-valve  how about "DeathMsg2" message here, with two strings(killer/victim name) and sprite path for custom icon? For example, Player1 killed Player2 by auto turret or some custom weapon, kill feed gonna be like
 `Player1's turret ♜ Player2` 
Then features like "kill assist" can be easily implemented by 3rd party plugin.

**Comment by sjain882** [comment]
I agree with PM here, gamedll fixes are better, but, if anything, wallbang icons definitely come first before any fancy stuff like this.

**Comment by MOCOLONI** [comment]
Sorry for bumping but I'd just like to say that I don't see why this shouldn't be a thing. If it was possible to add the (useless?) 'Money' feature into the scoreboard, why not also implement kill assists? After all and if I'm not wrong, I remember 'Counter-Strike Nexon' having it implemented in an update months after the game's release.

**Comment by CrazY770** [comment]
I believe make the game better doesn't mean copy everything from CS:GO. IMO CS 1.6 is fun as it's, copy things from a shit game would  make it a shit game as well.
